### PR TITLE
Add projects

### PR DESCRIPTION
**Pull request by StJudeWasHere**
## Description

Add projects.

## Proposed changes

Add the Project model and the needed routes to list, show, add, edit and delete projects.

## Type of change

- [ ] Improvement (change that improves an existing functionality)
- [x] Feature (New feature to the app)
- [ ] Chore

## Checklist

- [ ] I tested this pull request locally before submitting.
- [ ] I checked that this pull request does not conflict with other requests.
- [ ] I have made sure that this pull request follows the project's style and formatting guidelines.
- [ ] I performed unit/integration tests to ensure this pull request is functional.
- [ ] I have updated the documentation if necessary.

### CODE REVIEW

File: backend/app/Http/Controllers/ProjectController.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if (auth()->id() !== $project->user_id) {
+            return response()->json([
+                'success' => false,
+                'message' => 'Unauthorized to update this project.'
+            ], 403);
+        }
+
+        $validatedData = $request->validate([
+            'title' => 'sometimes|required|string|max:255',
+            'author' => 'sometimes|required|string|max:255',
```
Comment (CeeJayyy007) [inline backend/app/Http/Controllers/ProjectController.php]: **nitpick:** Awesome work with the PR. However, here you validated author field here but it is not present in project model

File: backend/app/Http/Controllers/ProjectController.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        ], 200);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, Project $project)
+    {
+        // 403 Forbidden. The user does not have permissions to edit this project.
+        if (auth()->id() !== $project->user_id) {
```
Comment (CeeJayyy007) [inline backend/app/Http/Controllers/ProjectController.php:64]: **nitpick:** could we also add a validation check for when `auth()->id()` might be null? Except if we do not envisage a scenario where an unauthenticated user would have access to the update project call then we are good.

**Review by CeeJayyy007 — APPROVED** [review]
LGTM! - However, I provided some comments that would be good if addressed. Awesome work @StJudeWasHere

File: backend/app/Http/Controllers/ProjectController.php (hunk 1 above)
Comment (StJudeWasHere) [reply backend/app/Http/Controllers/ProjectController.php]: You're right, I ended up removing the author field as we already have the user id. It seems I forgot to remove this validation, will fix.

File: backend/app/Http/Controllers/ProjectController.php (hunk 2 above)
Comment (StJudeWasHere) [reply backend/app/Http/Controllers/ProjectController.php:64]: I'd say that case scenario is out of the scope of this PR. If we want to allow anonymous projects, we would need to decide who and how they can update them. I'd say we allow only authenticated users for now, unless there's a clear path on how to do the unauthenticated thing.

**Pull request closed without merging**